modin/pandas/test/test_io.py

```diff
@@ -2029,15 +2029,19 @@ def test_read_parquet_5767(self, tmp_path, engine):
         # both Modin and pandas read column "b" as a category
         df_equals(test_df, read_df.astype("int64"))
 
-    def test_read_parquet_6855(self, tmp_path, engine):
+    @pytest.mark.parametrize("index", [False, True])
+    def test_read_parquet_6855(self, tmp_path, engine, index):
         if engine == "fastparquet":
             pytest.skip("integer columns aren't supported")
         test_df = pandas.DataFrame(np.random.rand(10**2, 10))
         path = tmp_path / "data"
         path.mkdir()
         file_name = "issue6855.parquet"
-        test_df.to_parquet(path / file_name, engine=engine)
+        test_df.to_parquet(path / file_name, index=index, engine=engine)
         read_df = pd.read_parquet(path / file_name, engine=engine)
+        if not index:
+            # In that case pyarrow cannot preserve index dtype
+            read_df.columns = pandas.Index(read_df.columns).astype("int64").to_list()
```

Comment:
We do the same in parquet_dispatcher.py. Am I right that we will have correct columns for the Modin DataFrame but wrong columns for partitions?

Reply:
The assigned columns will be propagated to partitions by `synchronize_labels` function.

Reply:
Should we do this in parquet_dispatcher.py like with sync_index flag?

Reply:
IIRC pyarrow handled this on its end by calling `.to_pandas` and using pandas metadata:

https://github.com/modin-project/modin/blob/56fb47e97cdb382cf0f9c20487c613a908c0580e/modin/core/storage_formats/pandas/parsers.py#L747-L758

Reply:
Why do we need then this line in the test if internal and external column labels match?

Reply:
In this case, the index is not saved, the `pyarrow` cannot obtain the same index type as the original dataframe had, but since I need to compare the original and read dataframe, I explicitly cast the type.

The fact that Modin reads the dataframe in this case with the another type of index is not Modin’s problem, the same behavior occurs in pandas. This is simply a consequence of the fact that due to the lack of metadata, the index type is lost.

> Why do we need then this line in the test if internal and external column labels match?

Both the internal type and the external type (of the columns) are the same, but they do not match what was in the original dataframe.

Reply:
I see, thanks for the explanation!